enhancements/ingress/customize-error-code-pages.md

```diff
@@ -0,0 +1,130 @@
+---
+title: customize-error-code-pages
+authors:
+  - "@miheer"
+reviewers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+approvers:
+  - "@Miciah"
+  - "@danehans"
+  - "@knobunc"
+creation-date: 2021-03-07 
+last-updated: 2020-03-07
+status: implementable 
+see-also:
+replaces:
+superseded-by:
+---
+# Customize error code paged returned by the haproxy router
+## Release Signoff Checklist
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+## Summary
+There is no supported and reasonable method to customize the HAProxy 2.x error page in OCP 4.x
+It's also important that the right page as per user requirements is served at the right moment.
+Example - 503 error page shall be returned if no pod is available. 
+But when the URI is wrong/does not exist, we should return 404 or a customized 404 page as per user requirements.
+Using this feature, user can provide custom http pages for 503 and 404 for the haproxy router by adding these pages
+in a configmap using --file attribute.
+## Motivation
+Users say 503 page is served even when the page is not found when it shall serve 404. 
+The reason that happens is because we provide only 503 in our haproxy template.
+
+### Goals
+1. Enable the cluster administrator to specify custom error code pages for haproxy router to serve them at the right moment.
```

Comment:
From the motivation and summary, it seems you can also add this goal: "Return a distinct HTTP 404 error page when the requested URI does not exist, instead of returning the HTTP 503 error page."

I think you can delete the phrase "at the right moment".  If the phrase refers to returning 404s, then that should be a separate goal.  If the phrase means something else, please clarify.